src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -360,16 +414,27 @@ export class ObjectExplorerService implements IObjectExplorerService {
 	public async createNewSession(providerId: string, connection: ConnectionProfile): Promise<azdata.ObjectExplorerSessionResponse> {
 		const provider = this._providers[providerId];
 		if (provider) {
+			// set the connection to wait for session
+			this._connectionsWaitingForSession.set(connection.id, true);
 			const result = await provider.createNewSession(connection.toConnectionInfo());
+			// some providers return a malformed create sessions responses which don't have a session id. We should throw an error in this case
+			if (!result?.sessionId) {
+				this.logService.error(`The session ID returned by provider "${providerId}" for connection "${connection.title}" is invalid.`);
+				throw new Error(nls.localize('objectExplorerSessionIdMissing', 'The session ID returned by provider "{0}" for connection "{1}" is invalid.', providerId, connection.title));
```

Comment:
nit: But generally we've been using single quotes for quoting text displayed to users

```suggestion
				throw new Error(nls.localize('objectExplorerSessionIdMissing', "No session ID returned by provider '{0}' for connection '{1}'.", providerId, connection.title));
```